client/my-sites/checkout/composite-checkout/lib/create-wpcom-account-before-transaction.ts

```diff
@@ -10,21 +10,31 @@ export async function createWpcomAccountBeforeTransaction(
 	const isJetpackUserLessCheckout = transactionCart.products.some(
 		( product ) => product.extra.isJetpackCheckout
 	);
+	const isAkismetUserLessCheckout = transactionCart.products.some(
+		( product ) => product.extra.isAkismetSitelessCheckout
+	);
 	const isGiftingCheckout = transactionCart.products.some(
 		( product ) => product.extra.isGiftPurchase
 	);
 
+	let signupFlowName;
```

Comment:
Thanks! This is much better than the growing ternary it replaces. I think the following block is pretty readable right now, but in my experience, over time if/else code like this will get muddied and mixed up with other conditions. Could we use an IIFE instead? Something like this:

```suggestion
	const signupFlowName = ( () => {
	if ( isJetpackUserLessCheckout ) {
		return 'jetpack-userless-checkout';
	}
	if ( isAkismetUserLessCheckout ) {
		return 'akismet-userless-checkout';
	}
	if ( isGiftingCheckout ) {
		return 'gifting-userless-checkout';
	}
	return 'onboarding-registrationless';
} )();
```

This way the priority of each value is clear (the conditions do not overlap right now but I am sure they will eventually), assignments will clearly be made in one place (often when a variable can be modified folks will start making assignments in different areas over time) and the whole thing could be moved to its own function if needed.

Reply:
Sure, that makes more sense to me as well. :)